packages/features/form-builder/schema.ts

```diff
@@ -289,7 +291,7 @@ export const fieldTypesSchemaMap: Partial<
       if (fields.length === 1) {
         const field = fields[0];
         if (variantSupportedFields.includes(field.type)) {
-          const schema = stringSchema;
+          const schema = field.required && !isPartialSchema ? nonEmptyString() : stringSchema;
```

Comment:
`required` field in name should be nonEmpty if it's marked required and it's not prefilling scenario(where it is okay to have no name or empty name)